### ISSUE TITLE
ROADMAP: Dropping .NET 4, EF 4, MVC 4

### ISSUE THREAD

**Issue by rockfordlhotka** [MarimerLLC/cslaforum#818]
I want to give everyone a heads-up. The next major release of CSLA (CSLA 6) will drop support for

* .NET 4
* .NET 4.5
* EF 4
* ASP.NET MVC 4

I expect this next major release will occur in the 6-12 month timeframe. Certainly when .NET 5 comes out, but possibly prior to that point.

I realize that some folks are still stuck on .NET 4.0 and 4.5. And I empathize, but I can't continue to support versions of Microsoft products that have been out of Microsoft support for years and years and years.

.NET 4.6.1 is the oldest version that I plan to support in CSLA 6, and that might change to 4.7.2 (which is the oldest version that's _really_ compatible with netstandard2.0). Though if I'm honest, what I _should_ do is target 4.8, because that's the last-ever version of .NET Framework and is probably what I should support as long as possible.

![image](https://user-images.githubusercontent.com/2333134/65700617-92192f00-e045-11e9-9f32-abf38c995d12.png)

**Comment by rockfordlhotka** [comment]
Yes, that is correct. As I understand it, .NET 5 is the unified replacement for all existing flavors of .NET.

**Comment by jonnybee** [comment]
I'd say drop EF5 support as well. 
Also checked my Azure AppServices and they are running in .NET 4.7.2 so I'd vote to stay on 4.7.2 at least until 4.8 is the default for Azure AppServices. 
See also: https://feedback.azure.com/forums/169385-web-apps/suggestions/37566262-upgrade-app-service-with-net-4-8

**Comment by rockfordlhotka** [comment]
@jonnybee That is a good point about EF 5, I've added a backlog item for that.

I do think I'll keep targeting .NET 4.6.1 until I'm forced to switch to 4.7. No sense causing anyone else pain until there's a pressing need to upgrade CSLA. Of course _that might happen_ as .NET 5 comes out 😄
